Sources/MapboxDirections/MapMatching/Tracepoint.swift

```diff
@@ -1,40 +1,47 @@
 import Foundation
 import CoreLocation
 
-/**
- A `Tracepoint` represents a location matched to the road network.
- */
-public class Tracepoint: Waypoint {
+public extension Match {
     /**
-     Number of probable alternative matchings for this tracepoint. A value of zero indicates that this point was matched unambiguously.
+     A tracepoint represents a location matched to the road network.
      */
-    public let countOfAlternatives: Int
-    
+    struct Tracepoint: Matchpoint, Equatable {
+        // MARK: Positioning the Waypoint
+        
+        /**
+         The geographic coordinate of the waypoint, snapped to the road network.
+         */
+        public var coordinate: CLLocationCoordinate2D
+        
+        /**
+         The straight-line distance from this waypoint to the corresponding waypoint in the `RouteOptions` or `MatchOptions` object.
+         
+         The requested waypoint is snapped to the road network. This property contains the straight-line distance from the original requested waypoint’s `DirectionsOptions.Waypoint.coordinate` property to the `coordinate` property.
+         */
+        public var correction: CLLocationDistance
+        
+        // MARK: Determining the Degree of Confidence
+        
+        /**
+         Number of probable alternative matchings for this tracepoint. A value of zero indicates that this point was matched unambiguously.
+         */
+        public var countOfAlternatives: Int
+        
+        public var name: String? // ??? matching response otherwise won't be able to decode waypoints names.
```

Comment:
If we don't supply a `name` in a `Tracepoint`, Map Matching response decoding won't pass this data to `RouteResponse` and in it's turn resulting `Waypoint`s won't have name too.  I see that `Tracepoint` is identical to `Route.Waypoint`, why do we need this struct at all?

Reply:
A waypoint (or tracepoint) name coming from the API isn’t actually that meaningful. Normally it’s just the name of the street that the waypoint got snapped to. If the application supplied a waypoint name as part of the request, that name is worth preserving. In the past, we’d override the API response’s waypoint names with the request’s waypoint names, where available. Now it’s up to the application to hang onto the request and associate the request’s waypoints with the response’s waypoints in some cases.

Reply:
Sorry, it's a bit misleading: "that name is worth preserving" and "it’s up to the application to hang onto the request and associate the request’s waypoints with the response’s waypoints" seems to be the opposite statements.
If we are to preserve waypoint name returned by the API (and apparently `Route.Waypoint` is), `Tracepoint` should have a `name`. Otherwise, `Waypoints` won't be decoded correctly as they depend on `Tracepoint` data: https://github.com/mapbox/mapbox-directions-swift/pull/388/files#diff-a2198eb412da79c9d8351c4e811b7dc6R63-R66